### PR TITLE
consequently use get_testfile

### PR DESCRIPTION
**Pull request by emanuel-schmid**
Changes proposed in this PR:

- use the `climada.test.get_testfile` function where applicable
- introduce a version distinction in the named function so that test datasets can be updated without breaking tests for previous versions.

### PR Author Checklist

- [x] Read the [Contribution Guide][contrib]
- [x] Correct target branch selected (if unsure, select `develop`)
- [x] Descriptive pull request title added
- [x] Source branch up-to-date with target branch
- [ ] [Documentation](https://climada-python.readthedocs.io/en/latest/guide/Guide_PythonDos-n-Donts.html#2.--Commenting-&-Documenting) updated
- [ ] [Tests][testing] updated
- [ ] [Tests][testing] passing
- [ ] No new [linter issues][linter]
- [ ] [Changelog](https://github.com/CLIMADA-project/climada_python/blob/main/CHANGELOG.md) updated

### PR Reviewer Checklist

- [ ] Read the [Contribution Guide][contrib]
- [ ] [CLIMADA Reviewer Checklist](https://climada-python.readthedocs.io/en/latest/guide/Guide_Reviewer_Checklist.html) passed
- [ ] [Tests][testing] passing
- [ ] No new [linter issues][linter]

[contrib]: https://github.com/CLIMADA-project/climada_python/blob/main/CONTRIBUTING.md
[testing]: https://climada-python.readthedocs.io/en/latest/guide/Guide_Continuous_Integration_and_Testing.html
[linter]: https://climada-python.readthedocs.io/en/stable/guide/Guide_Continuous_Integration_and_Testing.html#3.C.--Static-Code-Analysis

### CODE REVIEW

**Comment by chahank** [comment]
I think the method `get_test_file` is great. However, the current test file names in CLIMADA are very cryptic. Could we improve on that? Maybe we make better names, or the function allows for a name/description that is more useful?

**Comment by emanuel-schmid** [comment]
renaming the test datasets is a very good idea! However - I'd prefer to do it in another PR. This one doesn't touch the names, only the way to get to the data. _And_ it gets rid of the superfluous `climada/hazard/test/data/atl_prob_no_name.mat]` file!

File: climada/hazard/test/test_base.py [hunk 1]
```diff
@@ -41,7 +41,7 @@
 """
 Directory for writing (and subsequent reading) of temporary files created during tests.
 """
-HAZ_TEST_MAT :Path = Path(hazard_test.__file__).parent.joinpath('data', 'atl_prob_no_name.mat')
+HAZ_TEST_MAT :Path = get_test_file('atl_prob_no_name', file_format='matlab')
```
Comment (chahank) [inline climada/hazard/test/test_base.py:44]: Any reason to sometimes use the `HAZ_TEST_MAT :Path =` style and sometimes `HAZ_TEST_MAT =` ?

**Comment by chahank** [comment]
Can we also add a note in the developers guide to refer to this way of getting test data? Then over time it will harmonize.

File: climada/hazard/test/test_base.py (hunk 1 above)
Comment (emanuel-schmid) [reply climada/hazard/test/test_base.py:44]: No reason, except that no type annotation has been changed in this PR.

**Comment by emanuel-schmid** [comment]
I've added the note in the Developer Guide PR: https://github.com/CLIMADA-project/climada_python/commit/a8c671633effbb25bb91d2ab6a1fb00944fb6717

**Comment by chahank** [comment]
> I've added the note in the Developer Guide PR: [a8c6716](https://github.com/CLIMADA-project/climada_python/commit/a8c671633effbb25bb91d2ab6a1fb00944fb6717)

Excellent!

**Comment by chahank** [comment]
Please feel free to merge.

**Pull request merged**